pages/dkp/konvoy/1.6/addons/helmv2-to-v3-migration/index.md

```diff
@@ -30,6 +30,27 @@ kubectl -n kube-system delete deployment tiller-deploy
 
 This command shuts down Tiller gracefully and removes all its resources.
 
+## Upgrading Prometheus
+
+If you have the prometheus addon enabled and are upgrading it, you need to take additional steps to ensure it deploys successfully. You must delete the addon and then upgrade and deploy it.
+
+Before you begin the upgrade, run the following command to delete the addon:
+
+```bash
+kubectl delete addon prometheus -n kubeaddons
+helm2 delete --purge prometheus-kubeaddons
+```
+
+If you've already run the upgrade and deploying the prometheus addon has failed, run these commands:
+
+```bash
+kubectl delete addon prometheus -n kubeaddons
+helm2 delete --purge prometheus-kubeaddons
+konvoy deploy addons
+```
+
+These failures happen because there are limitations on the size of large configmaps in the migration of Helm v2 to Helm v3.
```

Comment:
```suggestion
The migration of a helm release from helm v2 to helm v3 adds additional metadata to the release data structure. This additional data causes the Prometheus release data to exceed the maximum size of a ConfigMap or Secret. Newer versions of the Prometheus chart reduces the amount of data held in this release structure, but the conversion will fail preventing the newer Prometheus chart installation from being reached. Removing the oversized data from kubernetes allows the replacement to continue.
```

Reply:
Aside: By putting sentences on their own lines, this allows simple git changes that have less overall impact, avoiding merge conflicts and making diffs clearer. Markdown will assemble these into proper paragraph format.